File: app/src/main/kotlin/com/kondenko/pocketwaka/api/model/stats/BestDay.java

package com.kondenko.pocketwaka.api.model.stats;

import android.content.Context;
import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import com.kondenko.pocketwaka.R;

import java.util.concurrent.TimeUnit;

public class BestDay implements Parcelable
{

    @SerializedName("created_at")
    @Expose
    public String createdAt;
    @SerializedName("date")
    @Expose
    public String date;
    @SerializedName("id")
    @Expose
    public String id;
    @SerializedName("modified_at")
    @Expose
    public Object modifiedAt;
    @SerializedName("total_seconds")
    @Expose
    public Integer totalSeconds;

    public String getHumanReadableTime(Context context) {
        String pattern = context.getString(R.string.stats_time_format);
        long hours = TimeUnit.SECONDS.toHours(totalSeconds);
        long minutes = TimeUnit.SECONDS.toMinutes(totalSeconds) - TimeUnit.HOURS.toMinutes(hours);
        return String.format(pattern, hours, minutes);
    }


    public final static Parcelable.Creator<BestDay> CREATOR = new Creator<BestDay>() {

        @SuppressWarnings({
            "unchecked"
        })
        public BestDay createFromParcel(Parcel in) {
            BestDay instance = new BestDay();
            instance.createdAt = ((String) in.readValue((String.class.getClassLoader())));
            instance.date = ((String) in.readValue((String.class.getClassLoader())));
            instance.id = ((String) in.readValue((String.class.getClassLoader())));
            instance.modifiedAt = in.readValue((Object.class.getClassLoader()));
            instance.totalSeconds = ((Integer) in.readValue((Integer.class.getClassLoader())));
            return instance;
        }

        public BestDay[] newArray(int size) {
            return (new BestDay[size]);
        }

    }
    ;

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(createdAt);
        dest.writeValue(date);
        dest.writeValue(id);
        dest.writeValue(modifiedAt);
        dest.writeValue(totalSeconds);
    }

    public int describeContents() {
        return  0;
    }

}

File: app/src/main/kotlin/com/kondenko/pocketwaka/data/stats/model/BestDay.kt
package com.kondenko.pocketwaka.data.stats.model

import android.content.Context
import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import com.kondenko.pocketwaka.R

import java.util.concurrent.TimeUnit

class BestDay : Parcelable {

    @SerializedName("created_at")
    @Expose
    var createdAt: String? = null

    @SerializedName("date")
    @Expose
    var date: String? = null

    @SerializedName("id")
    @Expose
    var id: String? = null

    @SerializedName("modified_at")
    @Expose
    var modifiedAt: Any? = null

    @SerializedName("total_seconds")
    @Expose
    var totalSeconds: Int? = null

    fun getHumanReadableTime(context: Context): String {
        val pattern = context.getString(R.string.stats_time_format)
        val hours = TimeUnit.SECONDS.toHours(totalSeconds!!.toLong())
        val minutes = TimeUnit.SECONDS.toMinutes(totalSeconds!!.toLong()) - TimeUnit.HOURS.toMinutes(hours)
        return String.format(pattern, hours, minutes)
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(createdAt)
        dest.writeValue(date)
        dest.writeValue(id)
        dest.writeValue(modifiedAt)
        dest.writeValue(totalSeconds)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {

        val CREATOR: Parcelable.Creator<BestDay> = object : Parcelable.Creator<BestDay> {
            override fun createFromParcel(`in`: Parcel): BestDay {
                val instance = BestDay()
                instance.createdAt = `in`.readValue(String::class.java.classLoader) as String
                instance.date = `in`.readValue(String::class.java.classLoader) as String
                instance.id = `in`.readValue(String::class.java.classLoader) as String
                instance.modifiedAt = `in`.readValue(Any::class.java.classLoader)
                instance.totalSeconds = `in`.readValue(Int::class.java.classLoader) as Int
                return instance
            }

            override fun newArray(size: Int): Array<BestDay?> {
                return arrayOfNulls(size)
            }

        }
    }

}
